spec/expression.dd

```diff
@@ -2878,55 +2901,57 @@ void foo()
         $(H5 $(LNAME2 is-identifier-equal, $(D is $(LPAREN)) $(I Type) $(I Identifier) $(D ==) $(I TypeSpecialization) $(D $(RPAREN))))
 
         $(P
-        The condition is satisfied if $(I Type) is semantically
-        correct and is the same as $(I TypeSpecialization).
-        The $(I Identifier) is declared to be either an alias of the
-        $(I TypeSpecialization) or, if $(I TypeSpecialization) is
-        dependent on $(I Identifier), the $(I TypeSpecialization) forms
-        a pattern from which the type of $(I Identifier) is deduced.
+        If *TypeSpecialization* is a type,
+        the condition is satisfied if $(I Type) is semantically correct and is
+        the same type as $(I TypeSpecialization).
+        $(I Identifier) is declared to be an alias of the
+        $(I TypeSpecialization).
         )
-
     $(SPEC_RUNNABLE_EXAMPLE_COMPILE
     ---
-    alias Bar = short;
+    const x = 5;
 
-    static if (is(Bar T == int))   // not satisfied, short is not int
+    static if (is(typeof(x) T == const int))   // satisfied, T is now defined
         alias S = T;
 
-    static assert(!is(T)); // T is not defined
+    static assert(is(T)); // T is in scope
+    pragma(msg, T); // const int
     ---
     )
 
         $(P
-        If $(I TypeSpecialization) is one of
-                $(D struct)
-                $(D union)
-                $(D class)
-                $(D interface)
-                $(D enum)
-                $(D __vector)
-                $(D function)
-                $(D delegate)
-                $(D const)
-                $(D immutable)
-                $(D inout)
-                $(D shared)
-                $(D module)
-                $(D package)
+        If $(I TypeSpecialization) is a type pattern involving
+        $(I Identifier), type deduction of $(I Identifier) is attempted
+        based on *Type*.
+        The condition is only satisfied if the type pattern is matched.
+        )
+    $(SPEC_RUNNABLE_EXAMPLE_COMPILE
+    ---
+    alias Foo = long*;
 
-        then the condition is satisfied if $(I Type) is one of those.
-        *TypeSpecialization* can also match the following keywords:
+    static if (is(Foo U == U*)) // Foo is matched against the pattern U*
+    {
+        U u;
+    }
+    static assert(is(U == long));
+    ---
+    )
+        $(P
+        If *TypeSpecialization* is a keyword, the condition is satisfied
+        in the same manner as the
+        $(RELATIVE_LINK2 is-type-equal, `is(Type == Keyword)` form).
+        *TypeSpecialization* can also be one of these keywords:
```

Comment:
I think these 3 keywords also work for the form without Identifier. `== return` does anyway.